_posts/2018-08-30-gsoc18-distancesearch.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+title: Improvements in distance search methods
+---
+
+We are pleased to announce another successful year of Google Summer of Code with [NumFOCUS][] organization. Thanks to [Richard Gowers][] and [Jonathan Barnoud][] for mentoring the GSoC students. This year, one of the projects, was to improve the performance of pairwise distance computations, which is used quite frequently in MDAnalysis in different forms. MDAnalysis v 0.19.0 and higher will include an internal function to speed up the neighbor search computations using automatic method selection. As a result, a user need not bother with the technical details of implementations of different search algorithms. A flexible functionality with easily extendible interface ``capped_distance`` is introduced in ``MDAnalysis.lib.distances`` for this purpose. Aside from the major highlight of ``capped_distance`` and its improvements, additional functionality of ``augment_coordinates`` is also implemented for any curious user to implement different neighbor search algorithm with periodic boundary conditions.
+
+One of the major bottleneck in various analysis routines in MDAnalysis (and typically in Molecular Dynamics studies) is the evaluation of pairwise distances among the particles. The primary problem revolves around fixed radius neighbor search algorithms. MDAnalysis offers a suite of algorithms including brute force method, tree-based binary search algorithms to solve such problems. While these methods are suitable for a variety of analysis functions using pairwise distances in MDAnalysis, one of the question was whether one can improve the performance of distance calculations using other established neighbor search methods.
+
+This question led to the inception of Google Summer of Code [project][] with [NumFOCUS][]. [Ayush Suhane][] completed the project and was able to demonstrate performance improvements for specific cases of distance selections, identification of bonds and Radial distribution function in the analysis module of MDAnalysis. More details on the commit history, PR's and blog posts can be found in the final [report][] submitted to GSoC. Real-time benchmarks for specific modules in MDAnalysis can be found [here](https://www.mdanalysis.org/benchmarks/). 
+
+The major highlight of the project is the introduction of ``capped_distance`` which allows automatic selection of methods based on predefined set of rules to evaluate pairs of atoms in the neighborhood of any particle. It allows a user-friendly interface for the developers to quickly implement any new algorithm throughout MDAnalysis modules. To test any new algorithm, one must comply with the following protocol:
```

Comment:
link to the docs – use the devdocs https://www.mdanalysis.org/mdanalysis/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.capped_distance

The major highlight of the project is the introduction of  [capped_distance()](https://www.mdanalysis.org/mdanalysis/documentation_pages/lib/distances.html#MDAnalysis.lib.distances.capped_distance), which allows automatic selection ...

Reply:
First, briefly describe the methods that are available and how they are chosen, and what their performances are. That's more interesting to the user than how to add new ones.